DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfigurationController.cs

```diff
@@ -116,5 +119,114 @@ public IActionResult EditCentreWebsiteDetails(EditCentreWebsiteDetailsViewModel
 
             return RedirectToAction("Index");
         }
+
+        [HttpGet]
+        [Route("RegistrationPrompts/{promptNumber}/Edit")]
+        public IActionResult EditRegistrationPrompt(int promptNumber)
+        {
+            var centreId = User.GetCentreId();
+
+            var customPrompt = customPromptsService.GetCustomPromptsForCentreByCentreId(centreId).CustomPrompts
+                .Single(cp => cp.CustomPromptNumber == promptNumber);
+
+            return View(new EditRegistrationPromptViewModel(customPrompt));
+        }
+
+        [HttpPost]
+        [Route("RegistrationPrompts/{promptNumber}/Edit")]
+        public IActionResult EditRegistrationPrompt(EditRegistrationPromptViewModel model, string action)
+        {
+            if (action.StartsWith("delete") && TryGetAnswerIndexFromAction(action, out var index))
+            {
+                return EditRegistrationPromptPostRemovePrompt(model, index);
+            }
+
+            return action switch
+            {
+                "save" => EditRegistrationPromptPostSave(model),
+                "addPrompt" => EditRegistrationPromptPostAddPrompt(model),
+                _ => RedirectToAction("Error", "LearningSolutions")
+            };
+        }
+
+        private bool TryGetAnswerIndexFromAction(string action, out int index)
+        {
+            return int.TryParse(action.Remove(0, 6), out index);
+        }
+
+        private IActionResult EditRegistrationPromptPostAddPrompt(EditRegistrationPromptViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                model.Options = NewlineSeparatedStringListHelper.SplitNewlineSeparatedList(model.OptionsString);
+                return View(model);
+            }
+
+            var (optionsString, options) =
+                NewlineSeparatedStringListHelper.AddStringToNewlineSeparatedList(model.OptionsString, model.Answer!);
+
+            if (model.OptionsString != null && optionsString.Length > 4000)
+            {
+                var remainingLength = 4000 - model.OptionsString.Length - 2;
+                ModelState.AddModelError
+                (
+                    nameof(EditRegistrationPromptViewModel.Answer),
+                    $"The complete list of answers must be less than 4000 characters. The new answer can be maximum {remainingLength} characters long."
+                );
+
+                ModelState.Remove(nameof(EditRegistrationPromptViewModel.Options));
+                model.Options = NewlineSeparatedStringListHelper.SplitNewlineSeparatedList(model.OptionsString);
+            }

```

Comment:
Should this check be before we check whether the modelstate is valid? (and maybe split into a private method so the controller is easier to scan over)

Reply:
I don't think so, we don't want to be doing all the adding it to the list logic if it is an invalid string e.g. too long..

I can pull it out into a private method though.

Reply:
I was more thinking of the flow of:
1. Check that the new length would not be > 4000 - if it is add a relevant error message to model errors.
2. If model state is not valid, show view.
3. If model state is valid, add a new option into the list.

Reply:
I was thinking more that it was pointless to append and check the length on an invalid prompt (empty or too long) but I guess there does exist the case where they are trying to add something too long that would also push it past the 4000. 

This does leave us with some awkward edge cases to the remaining space calculation like what if they have space for a 100 character prompt, but not the 150 character invalid prompt they were trying to add.